recording_external_revenue/models/recording_external_revenue_raw.py

```diff
@@ -55,37 +56,163 @@ class RecordingExternalRevenueRaw(models.Model):
+
+    def get_common_parameter_fields(self):
+        return [
+            "country",
+            "currency",
+            "fiscal_position",
+            "isrc",
+            "operation_date",
+            "partner",
+            "platform",
+            "product_external_catalog",
+            "product_external_catalog_reference",
+            "product_reference",
+            "recording_external_catalog",
+            "recording_external_catalog_reference",
+            "revenue_type",
+            "state",
+            "subplatform",
+            "tax",
+            "tax_base",
+            "upc",
+            "gross_amount_per_unit",
+        ]
+
+    @job
+    def convert(self):
+        self._check_not_already_converted()
+        new_revenue = self.make_new_revenue()
+        revenue = self.env["recording.external.revenue"].create(
+            dict(new_revenue._cache)
+        )
+        self.write({"revenue_id": revenue.id, "is_converted": True})
+        return revenue
+
+    def _check_not_already_converted(self):
+        for raw_revenue in self:
+            if raw_revenue.is_converted:
+                raise ValidationError(
+                    _(
+                        "The raw revenue {} is already converted. "
+                        "It can not be converted twice."
+                    ).format(raw_revenue.display_name)
+                )
+
+    def make_new_revenue(self):
         revenue = self.env["recording.external.revenue"].new()
+        self[0]._execute_direct_mapping(revenue)
+        self[0]._execute_advanced_mapping(revenue)
+        self._execute_aggregated_fields_mapping(revenue)
+        return revenue
 
+    def _execute_direct_mapping(self, revenue):
         for field in self.get_direct_mapping_fields():
             revenue[field] = self[field]
 
-        revenue.partner_id = self._map_partner()
-        revenue.country_id = self._map_country()
-        revenue.state_id = self._map_country_state()
-        revenue.platform_id = self._map_platform()
-        revenue.subplatform_id = self._map_subplatform()
-        revenue.product_id = self._map_product()
-        revenue.recording_id = self._map_recording()
-        revenue.tax_id = self._map_tax()
-        revenue.currency_id = self._map_currency()
-        return revenue
-
     def get_direct_mapping_fields(self):
         return [
-            "commission_amount",
+            "company_id",
             "fiscal_position",
-            "gross_amount",
             "gross_amount_per_unit",
-            "net_amount",
             "operation_date",
             "period_end_date",
             "period_start_date",
             "quantity",
             "tax_base",
         ]
 
+    def _execute_aggregated_fields_mapping(self, revenue):
+        aggregated_fields = self.get_aggregated_fields()
+        for field in aggregated_fields:
+            revenue[field] = sum(r[field] or 0 for r in self)
+
+    def get_aggregated_fields(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-recording-addons/pullRequest?prid=5620550)